test/Datadog.AutoInstrumentation.NativeLoader.Tests/string_test.cpp

```diff
@@ -0,0 +1,76 @@
+#include "gtest/gtest.h"
+#include "../../src/Datadog.AutoInstrumentation.NativeLoader/string.h"
+
+TEST(string, ToString)
+{
+    EXPECT_TRUE("Normal String" == ToString(std::string("Normal String")));
+    EXPECT_TRUE("\tNormal String\0" == ToString(std::string("\tNormal String\0")));
+
+    EXPECT_TRUE("Char* String" == ToString("Char* String"));
+    EXPECT_TRUE("\tChar* String\0" == ToString("\tChar* String\0"));
+
+    EXPECT_TRUE("Wide String" == ToString(WStr("Wide String")));
+    EXPECT_TRUE("\tWide String\0" == ToString(WStr("\tWide String\0")));
+
+    EXPECT_TRUE("42" == ToString(42));
+
+    EXPECT_TRUE("LPTSTR String" == ToString(TEXT("LPTSTR String")));
+    EXPECT_TRUE("\tLPTSTR String\0" == ToString(TEXT("\tLPTSTR String\0")));
+}
+
+TEST(string, ToWSTRING)
+{
+    EXPECT_TRUE(WStr("Normal String") == ToWSTRING(std::string("Normal String")));
+    EXPECT_TRUE(WStr("\tNormal String\0") == ToWSTRING(std::string("\tNormal String\0")));
+
+    EXPECT_TRUE(WStr("42") == ToWSTRING(42));
+}
+
+TEST(string, Trim)
+{
+    EXPECT_TRUE(WStr("WideString") == Trim(WStr("               WideString")));
+    EXPECT_TRUE(WStr("WideString") == Trim(WStr("WideString               ")));
+    EXPECT_TRUE(WStr("WideString") == Trim(WStr("               WideString               ")));
+
+    EXPECT_TRUE(WStr("Wide String") == Trim(WStr("               Wide String")));
+    EXPECT_TRUE(WStr("Wide String") == Trim(WStr("Wide String               ")));
+    EXPECT_TRUE(WStr("Wide String") == Trim(WStr("               Wide String               ")));
+
+    //
+
+    EXPECT_TRUE("NormalString" == Trim("               NormalString"));
+    EXPECT_TRUE("NormalString" == Trim("NormalString               "));
+    EXPECT_TRUE("NormalString" == Trim("               NormalString               "));
+
+    EXPECT_TRUE("Normal String" == Trim("               Normal String"));
+    EXPECT_TRUE("Normal String" == Trim("Normal String               "));
+    EXPECT_TRUE("Normal String" == Trim("               Normal String               "));
```

Comment:
`EXPECT_TRUE("Normal String" == Trim(" Normal String  \n  "));` will fail, but it should succeed. See note in the implementation.

Reply:
Fixed and test case added in b63d9a9

Reply:
✔️